bootstrap

```diff
@@ -157,7 +157,7 @@ fi
 find teuthology -name '*.pyc' -exec rm {} \;
 
 # Install teuthology in editable mode
-./$VENV/bin/pip install --editable .
+./$VENV/bin/pip install --editable '.[test]'
```

Comment:
I've spent about five minutes searching the pip/PEP docs for wtf .[test] might mean (and still haven't found it).  Could you add a comment with at least the name of the concept for the next poor soul?

Reply:
This is similar to doing `pip install teuthology[test]`, and indicates we want to install [optional dependencies](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies) as specified [in setup.cfg](https://github.com/ceph/teuthology/pull/1737/files#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52R99-R110). I can add a comment to that effect